app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,60 @@
+class VendorsController < ApplicationController
+
+  def index
+    @vendors = Vendor.all
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
+    @beg_month = Time.now.beginning_of_month
+    @end_month = Time.now.end_of_month
+    @all_sales = @vendor.sales.where(purchase_time: (@beg_month..@end_month))
+  end
+
+  def new
+    @vendor = Vendor.new
+    @market = Market.find(params["format"])
+  end
+
```

Comment:
Instead of doing things with 'format' in the params, you could use the relationship you set up between vendor and market and do @vendor.market when you need to refer to it's market. Seems simpler and then the route won't have a weird number after it in the url bar. Same goes for the other places you use 'format'.